### PR TITLE
Fix unhashable types in the ReqMgr2 request API with mask

### LINKED ISSUE #9832: ReqMgr2 'request' REST API fails with mask=ScramArch
**Impact of the bug**
ReqMgr2

**Describe the bug**
Similar issue to: https://github.com/dmwm/WMCore/issues/9831

Making a call to reqmgr2 `request` API with this query string:
https://cmsweb.cern.ch/reqmgr2/data/request?status=acquired&mask=ScramArch

fails on the server side and return a `500 Internal Server Error`.

While other masks work just fine, e.g.:
https://cmsweb.cern.ch/reqmgr2/data/request?status=acquired&mask=OutputDatasets

**How to reproduce it**
See urls above, GET request

**Expected behavior**
The server should properly handle these requests.

**Additional context and error message**
reqmgr2 log traceback is:
```
[26/Feb/2021:01:07:45]  SERVER OTHER ERROR __builtins__.TypeError e8607f347986e4a31d418abb9ad920e3 (unhashable type: 'list')
[26/Feb/2021:01:07:45]    Traceback (most recent call last):
[26/Feb/2021:01:07:45]      File "/data/srv/HG2103e/sw/slc7_amd64_gcc630/cms/reqmgr2/1.4.6.pre7/lib/python2.7/site-packages/WMCore/REST/Server.py", line 726, in default
[26/Feb/2021:01:07:45]        return self._call(RESTArgs(list(args), kwargs))
[26/Feb/2021:01:07:45]      File "/data/srv/HG2103e/sw/slc7_amd64_gcc630/cms/reqmgr2/1.4.6.pre7/lib/python2.7/site-packages/WMCore/REST/Server.py", line 807, in _call
[26/Feb/2021:01:07:45]        obj = apiobj['call'](*safe.args, **safe.kwargs)
[26/Feb/2021:01:07:45]      File "/data/srv/HG2103e/sw/slc7_amd64_gcc630/cms/reqmgr2/1.4.6.pre7/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 369, in get
[26/Feb/2021:01:07:45]        result = self._mask_result(mask, result)
[26/Feb/2021:01:07:45]      File "/data/srv/HG2103e/sw/slc7_amd64_gcc630/cms/reqmgr2/1.4.6.pre7/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 252, in _mask_result
[26/Feb/2021:01:07:45]        self._maskTaskStepChain(masked_result[req_name], req_info, "Task", mask_key)
[26/Feb/2021:01:07:45]      File "/data/srv/HG2103e/sw/slc7_amd64_gcc630/cms/reqmgr2/1.4.6.pre7/lib/python2.7/site-packages/WMCore/ReqMgr/Service/Request.py", line 237, in _maskTaskStepChain
[26/Feb/2021:01:07:45]        masked_dict[mask_key] = list(set(masked_dict[mask_key]))
[26/Feb/2021:01:07:45]    TypeError: unhashable type: 'list'
```

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #9832 

#### Status
ready

#### Description
Fixes the way we extract a mask of keys from the request dictionary, such that it uses the properly implemented and tested `RequestInfo` object, being fixed in: https://github.com/dmwm/WMCore/pull/10311

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
Logical fix made in (to be merged before this one): https://github.com/dmwm/WMCore/pull/10311

#### External dependencies / deployment changes
None

### CODE REVIEW

**Comment by amaltaro** [comment]
Tested in my VM and it seems to be working fine.

File: src/python/WMCore/ReqMgr/Service/Request.py [hunk 1]
```diff
@@ -207,51 +208,27 @@ def validate(self, apiobj, method, api, param, safe):
...
+            request dictionary as a value
+        :return: updates the result object in place and returns it (dict)
+        """
 
         if len(mask) == 1 and mask[0] == "DAS":
             mask = ReqMgrConfigDataCache.getConfig("DAS_RESULT_FILTER")["filter_list"]
 
         if len(mask) > 0:
             masked_result = {}
             for req_name, req_info in result.items():
```
Comment (todor-ivanov) [inline src/python/WMCore/ReqMgr/Service/Request.py:226]: Following the good trend from the rest of this PR of migrating from snake_case to camelCase names, I would suggest to take the opportunity and do it fro every variable in this function + the function name itself, provided we know all the places this one is called from.

**Review by todor-ivanov — CHANGES_REQUESTED** [review]
Thanks, @amaltaro, I left one comment inline. All the rest looks good to me.

File: src/python/WMCore/ReqMgr/Service/Request.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/ReqMgr/Service/Request.py:226]: Don't you think it's kind of pointless to convert variables to camel case only in one function?
I fully agree with converting the whole module to be conformant with our pylintrc standards, but that would require a whole new validation to make sure no mistakes were made.

**Comment by amaltaro** [comment]
@todor-ivanov can you please have another look at this PR? Second commit contains the changes you suggested.

File: src/python/WMCore/ReqMgr/Service/Request.py (hunk 1 above)
Comment (todor-ivanov) [reply src/python/WMCore/ReqMgr/Service/Request.py]: I wouldn't dare to suggest a change for the whole module, but for sure I agree with you it is worth a whole separate issue.

**Review by todor-ivanov — APPROVED** [review]
Thanks @amaltaro  Looks good to me now .

**Pull request merged**